pkg/api/v1/types.go

```diff
@@ -2301,6 +2307,10 @@ type DeleteOptions struct {
 	// specified type will be used.
 	// Defaults to a per object value if not specified. zero means delete immediately.
 	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
+
+	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
+	// returned.
+	Preconditions Preconditions `json:"preconditions"`
```

Comment:
I'm fine with the general idea, and with the name of the field. 

However, does a nested struct work with the automagic option converter?

Otherwise, same comments as on api/types.go.